testsuite/tests/singlecluster/extensions/plan_policy/test_plan_policy.py

```diff
@@ -0,0 +1,118 @@
+"""Tests for PlanPolicy functionality including authentication and rate limiting."""
+
+import pytest
+
+from testsuite.httpx.auth import HttpxOidcClientAuth
+from testsuite.kuadrant.extensions.plan_policy import PlanPolicy, Plan
+
+
+pytestmark = [pytest.mark.kuadrant_only, pytest.mark.authorino, pytest.mark.extensions]
+
+PLANS = {
+    "gold": Plan(
+        tier="gold",
+        predicate='has(auth.identity) && auth.identity.planID == "gold"',
+        limits={"custom": [{"limit": 3, "window": "20s"}]},
```

Comment:
is there a user-story you're getting these limits from? If no, I'd put a bit more realistic plan limits here